### PR TITLE
pkg, strict version

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

set requirements for `lighnting_app` strict to the generated version

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @borda

### CODE REVIEW

**Review by krshrimali — APPROVED** [review]
@Borda - Just confirming, is the output below expected? Is there a reason why `==1.8.*` (for PL) is repeated twice?

(value of [reqs](https://github.com/Lightning-AI/lightning/blob/master/.actions/setup_tools.py#L514))

Before this PR: (on `master`)

```
['pytorch_lightning>=1.6.5, ==1.8.*', 'lightning_app>=0.5.2, ==0.7.*']
```

After this PR:

```
['pytorch_lightning==1.8.*,>=1.6.5, ==1.8.*', 'lightning_app==0.7.*,==0.7.0dev,>=0.5.2, ==0.7.0dev']
```

**Comment by Borda** [comment]
No, the output after is:
```
['pytorch_lightning>=1.6.5, ==1.8.*', 'lightning_app>=0.5.2, ==0.7.0dev']
```
you just need to run it once :otter:

**Review by krshrimali — APPROVED** [review]
Discussed offline, it accumulated the original file (hence the duplicates). LGTM 🚀

**Review by kaushikb11 — APPROVED** [review]

**Comment by Borda** [comment]
interesting:
```
ERROR: Could not find a version that satisfies the requirement lightning_app==0.7.0,>=0.5.2 (from lightning) (from versions: 0.5.1, 0.5.2, 0.5.3, 0.5.4, 0.5.5, 0.5.6, 0.5.7, 0.6.0rc0, 0.6.0, 0.6.1, 0.7.0.dev0)
ERROR: No matching distribution found for lightning_app==0.7.0,>=0.5.2
Error: Process completed with exit code 1.
```
seems that for `==0.7.0` is not satisfied with `0.7.0.dev0` even with `--pre` argument

File: .actions/setup_tools.py [hunk 1]
```diff
@@ -500,18 +500,22 @@ def _download_frontend(root: str = _PROJECT_ROOT):
...
 
-def _relax_require_versions(source_dir: str = "src", req_dir: str = "requirements") -> None:
+def _relax_require_versions(
+    source_dir: str = "src", req_dir: str = "requirements", strict_pkgs: Sequence[str] = ("lightning_app",)
+) -> None:
     """Parse the base requirements and append  as version adjustments if needed `pkg>=X1.Y1.Z1,==X2.Y2.*`.
 
     >>> _relax_require_versions("../src", "../requirements")
     """
+    strict_pkgs = strict_pkgs or tuple()
```
Comment (carmocca) [inline .actions/setup_tools.py:510]: Why do this? strict_pkgs does not allow None

**Pull request merged**